### PR TITLE
zeromq: add version 4.3.5, change license, add package_type

### PR DESCRIPTION
**Pull request by toge**
Specify library name and version:  **zeromq/***

- license has been changed since 4.3.5
- add package_type
- add package_description
- update dependencies

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/zeromq/all/conandata.yml [hunk 1]
```diff
@@ -9,14 +12,34 @@ sources:
...
+    - patch_file: "patches/0004-cmake-minimum-required-first-4.3.5.patch"
+      patch_description: "exchange positions between project and cmake_minimum_required"
+      patch_type: "portability"
   "4.3.4":
     - patch_file: "patches/0003-rpath-macos-4.3.3.patch"
+      patch_description: "disable rpath on macOS"
+      patch_type: "portability"
     - patch_file: "patches/0004-cmake-minimum-required-first.patch"
+      patch_description: "exchange positions between project and cmake_minimum_required"
+      patch_type: "portability"
```
Comment (SpaceIm) [inline recipes/zeromq/all/conandata.yml:25]: please keep all previous patches in this new versions

File: recipes/zeromq/all/conandata.yml [hunk 2]
```diff
@@ -9,14 +12,34 @@ sources:
...
   "4.3.3":
     - patch_file: "patches/0003-rpath-macos-4.3.3.patch"
+      patch_description: "disable rpath on macOS"
```
Comment (SpaceIm) [inline recipes/zeromq/all/conandata.yml:28]: What do you mean? This patch doesn't disable `rpath`.

Comment (toge) [reply recipes/zeromq/all/conandata.yml:28]: @SpaceIm
Thank you for youre comment.
Now I understand.

**Review by RubenRBS — DISMISSED** [review]
Thanks for the PR!

File: recipes/zeromq/all/conanfile.py [hunk 3]
```diff
@@ -46,6 +46,8 @@ def export_sources(self):
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
+        if Version(self.version) >= "4.3.5":
+            self.license = "MPL-2.0"
```
Comment (jwillikers) [inline recipes/zeromq/all/conanfile.py:50]:
```suggestion
        if Version(self.version) >= "4.3.5":
            self.license = "MPL-2.0"
        else:
            self.license = "DocumentRef-ZeroMQ:LicenseRef-LGPL-3.0-or-later-ZeroMQ-Linking-Exception"
```

File: recipes/zeromq/all/conanfile.py [hunk 4]
```diff
@@ -114,6 +121,7 @@ def build(self):
 
     def package(self):
         copy(self, "COPYING*", src=self.source_folder, dst=os.path.join(self.package_folder, "licenses"))
+        copy(self, "LICENSE", src=self.source_folder, dst=os.path.join(self.package_folder, "licenses"))
```
Comment (jwillikers) [inline recipes/zeromq/all/conanfile.py:124]:
```suggestion
        if Version(self.version) >= "4.3.5":
            copy(self, "LICENSE", self.source_folder, os.path.join(self.package_folder, "licenses"))
        else:
            copy(self, "COPYING*", self.source_folder, os.path.join(self.package_folder, "licenses"))
```

**Review by jwillikers — APPROVED** [review]
Thanks!

**Review by danimtb — APPROVED** [review]

File: recipes/zeromq/all/conandata.yml [hunk 5]
```diff
@@ -9,14 +12,37 @@ sources:
     url: "https://github.com/zeromq/libzmq/releases/download/v4.3.2/zeromq-4.3.2.tar.gz"
     sha256: "ebd7b5c830d6428956b67a0454a7f8cbed1de74b3b01e5c33c5378e22740f763"
 patches:
+  "4.3.5":
+    - patch_file: "patches/0003-rpath-macos-4.3.5.patch"
+      patch_description: "hardcoded full install path on local machine"
```
Comment (SpaceIm) [inline recipes/zeromq/all/conandata.yml:17]:
```suggestion
      patch_description: "hardcoded full install path on local machine"
```

well it's the opposite, this patch removes hardcoded full install path

**Pull request merged**